datafusion/physical-plan/src/joins/hash_join.rs

```diff
@@ -1023,11 +1048,9 @@ pub fn build_equal_condition_join_indices<T: JoinHashMapType>(
     random_state: &RandomState,
     null_equals_null: bool,
     hashes_buffer: &mut Vec<u64>,
-    filter: Option<&JoinFilter>,
-    build_side: JoinSide,
-    deleted_offset: Option<usize>,
-    fifo_hashmap: bool,
-) -> Result<(UInt64Array, UInt32Array)> {
+    limit: usize,
+    offset: JoinHashMapOffset,
+) -> Result<(UInt64Array, UInt32Array, Option<JoinHashMapOffset>)> {
     let keys_values = probe_on
```

Comment:
I believe the probe hashes are being recalculated unnecessarily if the probe batch is not completed yet. To improve efficiency, an offset can be used.

Reply:
Fixed, now hashes calculated in `fetch_probe_batch`, once per multiple possible iterations.